src/Microsoft.SqlTools.ServiceLayer/Agent/AgentService.cs

```diff
@@ -1211,6 +1219,215 @@ public static XmlDocument CreateJobXmlDocument(string svrName, string jobName)
             return new Tuple<SqlConnectionInfo, DataTable, ServerConnection>(sqlConnInfo, dt, serverConnection);
         }
 
+                // Notebook helper function to execute queires to be later replaced by QueryExecutionService method.
+        public DataSet executeQuery(ConnectionInfo connInfo, string SqlQuery)
+        {
+            DataSet resultDataSet = new DataSet();
+            using (SqlConnection connection = new SqlConnection(ConnectionService.BuildConnectionString(connInfo.ConnectionDetails)))
+            {
+                connection.Open();
+                using( SqlCommand SqlQueryCommand = new SqlCommand(SqlQuery, connection))
+                {
+                    SqlDataAdapter SqlCommandAdapter = new SqlDataAdapter(SqlQueryCommand);
+                    SqlCommandAdapter.Fill(resultDataSet);
+                }
+            }
+            return resultDataSet;
+        }
+
+
+
+        internal async Task HandleAgentNotebooksRequest(AgentNotebooksParams parameters, RequestContext<AgentNotebooksResult> requestContext)
+        {
+            await Task.Run(async () =>
+            {
+                try
+                {
+                    var result = new AgentNotebooksResult();
+                    ConnectionInfo connInfo;
+                    ConnectionServiceInstance.TryFindConnection(
+                        parameters.OwnerUri,
+                        out connInfo);
+
+                    if (connInfo != null)
+                    {
```

Comment:
If connInfo is null you never send a response and thus will time out on the ADS side. You should send an error in that case.